docs/reference/components/sources/exec.cue

```diff
@@ -0,0 +1,306 @@
+	}
+
+	configuration: {
+		mode: {
+			description: "The type of exec mechanism."
+			required:    true
+			warnings: []
+			type: string: {
+				enum: {
+					scheduled: "Scheduled exec mechanism."
+					streaming: "Streaming exec mechanism."
+				}
+				syntax: "literal"
+			}
+		}
+		command: {
+			common:      false
+			required:    false
+			description: "The command to be run."
+			warnings: []
+			type: string: {
+				default: null
+				examples: ["echo", "./myscript.sh"]
+				syntax: "literal"
+			}
+		}
+		arguments: {
+			common:      false
+			description: "Array of any arguments to pass to the command."
+			required:    false
+			type: array: {
+				default: null
+				items: type: string: {
+					examples: ["Hello World!"]
+					syntax: "literal"
+				}
+			}
+		}
+		exec_interval_secs: {
+			common:        false
+			description:   "The interval in seconds between scheduled command runs."
+			relevant_when: "mode = `scheduled`"
+			required:      false
+			type: uint: {
+				default: 60
+				unit:    "seconds"
+			}
+		}
+		event_per_line: {
+			common:        false
+			description:   "Determine if events should be generated per line."
+			relevant_when: "mode = `scheduled`"
+			required:      false
+			type: bool: default: true
+		}
+		exec_duration_millis_key: {
+			category:      "Context"
+			common:        false
+			description:   "The key name added to each event representing the duration in millis that the command took to run."
+			relevant_when: "mode = `scheduled`"
+			required:      false
+			warnings: []
+			type: string: {
+				default: "exec_duration_millis"
+				syntax:  "literal"
+			}
+		}
+		respawn_on_exit: {
+			common:        false
+			description:   "Determine if a streaming command should be restarted if it exits."
+			relevant_when: "mode = `streaming`"
+			required:      false
+			type: bool: default: true
+		}
+		respawn_interval_secs: {
+			common:        false
+			description:   "The interval in seconds between restarting streaming commands if needed."
+			relevant_when: "mode = `streaming`"
+			required:      false
+			warnings: []
+			type: uint: {
+				default: 60
+				unit:    "seconds"
+			}
+		}
+		current_dir: {
+			common:      false
+			required:    false
+			description: "The directory from within which to run the command."
+			warnings: []
+			type: string: {
+				default: null
+				syntax:  "literal"
+			}
+		}
+		include_stderr: {
+			common:      false
+			description: "Include the output of stderr when generating events."
+			required:    false
+			type: bool: default: false
+		}
+		host_key: {
+			category:    "Context"
+			common:      false
+			description: "The key name added to each event representing the current host. This can also be globally set via the [global `host_key` option][docs.reference.configuration.global-options#host_key]."
+			required:    false
+			warnings: []
+			type: string: {
+				default: "host"
+				syntax:  "literal"
+			}
+		}
+
+		pid_key: {
```

Comment:
I realize there is some precedent for exposing these as configurable options to optionally add metadata, but since the introduction of remap, we've been moving more towards just setting the keys and then letting users use `remap` to rename them or delete them if they don't want them. What would you think of just always setting `pid`, `exit_status`, etc.?

Reply:
Agreed, it seems much cleaner to do it that way. I have now removed the optional keys and made them part of the log events all the time.